yash-builtin/src/command/syntax.rs

```diff
@@ -0,0 +1,204 @@
+                }
+            );
+        });
+    }
+
+    #[test]
+    fn invoke_with_p_option() {
+        let env = Env::new_virtual();
+        let result = parse(&env, Field::dummies(["-p", "foo"]));
+
+        assert_matches!(result, Ok(Command::Invoke(invoke)) => {
+            assert_eq!(invoke.fields, Field::dummies(["foo"]));
+            assert_eq!(
+                invoke.search,
+                Search {
+                    standard_path: true,
+                    categories: Category::Builtin | Category::ExternalUtility
+                }
+            );
+        });
+    }
+
+    #[test]
+    fn identify_without_options() {
+        let env = Env::new_virtual();
+        let result = parse(&env, Field::dummies(["-v", "foo"]));
+
+        assert_matches!(result, Ok(Command::Identify(identify)) => {
+            assert_eq!(identify.names, Field::dummies(["foo"]));
+            assert_eq!(
+                identify.search,
+                Search {
+                    standard_path: false,
+                    categories: EnumSet::all()
+                }
+            );
+            assert!(!identify.verbose);
+        });
+    }
+
+    #[test]
+    fn identify_with_p_option() {
+        let env = Env::new_virtual();
+        let result = parse(&env, Field::dummies(["-v", "-p", "foo"]));
+
+        assert_matches!(result, Ok(Command::Identify(identify)) => {
+            assert_eq!(identify.names, Field::dummies(["foo"]));
+            assert_eq!(
+                identify.search,
+                Search {
+                    standard_path: true,
+                    categories: EnumSet::all()
+                }
+            );
+            assert!(!identify.verbose);
+        });
+    }
+
+    #[test]
+    fn verbosely_identify_without_options() {
+        let env = Env::new_virtual();
+        let result = parse(&env, Field::dummies(["-V", "bar"]));
+
+        assert_matches!(result, Ok(Command::Identify(identify)) => {
+            assert_eq!(identify.names, Field::dummies(["bar"]));
+            assert_eq!(
+                identify.search,
+                Search {
+                    standard_path: false,
+                    categories: EnumSet::all()
+                }
+            );
+            assert!(identify.verbose);
+        });
+    }
+
+    // This ordering is not specified by POSIX, but it is consistent with the
+    // older versions of yash.
+    #[test]
+    #[allow(non_snake_case)]
+    fn last_specified_option_wins_between_v_and_V() {
+        let env = Env::new_virtual();
+
+        let result = parse(&env, Field::dummies(["-V", "-v", "baz"]));
+        assert_matches!(result, Ok(Command::Identify(identify)) => {
+            assert!(!identify.verbose);
+        });
+
+        let result = parse(&env, Field::dummies(["-v", "-V", "baz"]));
+        assert_matches!(result, Ok(Command::Identify(identify)) => {
+            assert!(identify.verbose);
+        });
+    }
```

Comment:
The currently possible error case is only `Error::CommonError`, which only occurs as a result of `parse_arguments`. Since test cases for `parse_arguments` already tests for the error cases, I don't think it would be very beneficial to add more tests.